src/core/atomic.d

```diff
@@ -264,20 +264,35 @@ in (atomicPtrIsProperlyAligned(here), "Argument `here` is not properly aligned")
 }
 
 /**
- * Stores 'writeThis' to the memory referenced by 'here' if the value
- * referenced by 'here' is equal to 'ifThis'.  This operation is both
- * lock-free and atomic.
+ * Performs the atomic operation known as "compare-and-swap" (CAS).
+ *
+ * Compare-and-swap is essentially the following operation:
+ * ---
+ * if (*here == ifThis)
+ * {
+ *   *here = writeThis;
+ *   return true;
+ * }
+ * else
+ *   return false;
+ * ---
+ * This operation is both lock-free and atomic.
  *
  * Params:
  *  here      = The address of the destination variable.
  *  writeThis = The value to store.
  *  ifThis    = The comparison value.
  *
  * Returns:
- *  true if the store occurred, false if not.
+ *  Whether the swap occured or not.
+ *
+ * See_Also:
+ *  https://en.wikipedia.org/wiki/Compare-and-swap
  */
-bool cas(MemoryOrder succ = MemoryOrder.seq,MemoryOrder fail = MemoryOrder.seq,T,V1,V2)(T* here, V1 ifThis, V2 writeThis) pure nothrow @nogc @trusted
-    if (!is(T == shared) && is(T : V1))
+bool cas
+    (MemoryOrder succ = MemoryOrder.seq, MemoryOrder fail = MemoryOrder.seq, T, V1, V2)
+    (T* here, V1 ifThis, V2 writeThis) pure nothrow @nogc @trusted
+    if (!is(T == shared) && is(V1 : T))
```

Comment:
hard to tell from the diff if you made functional changes or just reformat the existing code...
i'm having to audit this carefully, but i probably don't need to?
if this sort of refactor is fair game, i'll definitely just change this right back next time i touch it ;)